cli/cmd/validate.go

```diff
@@ -0,0 +1,135 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/cloudquery/cloudquery/cli/internal/auth"
+	"github.com/cloudquery/cloudquery/cli/internal/specs/v0"
+	"github.com/cloudquery/plugin-pb-go/managedplugin"
+	"github.com/cloudquery/plugin-pb-go/pb/plugin/v3"
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+)
+
+const (
+	validateShort   = "Validate spec(s) files"
+	validateExample = `# Validate spec(s) files
+cloudquery validate ./directory
+# Validate specs from directories and files
+cloudquery validate ./directory ./aws.yml ./pg.yml
+`
+)
+
+func newCmdValidate() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "validate [files or directories]",
+		Short:   validateShort,
+		Long:    validateShort,
+		Example: validateExample,
+		Args:    cobra.MinimumNArgs(1),
+		RunE:    validate,
+		Hidden:  true,
+	}
+	return cmd
+}
+
+func validate(cmd *cobra.Command, args []string) error {
+	cqDir, err := cmd.Flags().GetString("cq-dir")
+	if err != nil {
+		return err
+	}
+
+	ctx := cmd.Context()
+	log.Info().Strs("args", args).Msg("Loading spec(s)")
+	fmt.Printf("Loading spec(s) from %s\n", strings.Join(args, ", "))
```

Comment:
Should this be here as you have the log line before?

Reply:
So I think we need both, see https://github.com/cloudquery/cloudquery/blob/c6e06aa2fb25010e0266538200dcb7212aa17248/cli/cmd/sync.go#L99

one goes to the log, the other shows up in the console via `stdout`